core/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java

```diff
@@ -221,4 +222,24 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         return "delete {[" + index + "][" + type + "][" + id + "]}";
     }
+
+    // Override these methods from ReplicationAction, these are things where we are storing our state in the request object (which we really
+    // shouldn't do). Once the transport client goes away we can move away from making these available, but in the meantime these are
+    // dangerous to set or use because the DeleteRequest object will always be wrapped in a bulk request envelope, which is where these
+    // *should* be set.
```

Comment:
Please promote these to legitimate Javadocs on each method so they show up in IDEs, etc.